crates/ruff_linter/src/rules/ruff/rules/missing_fstring_syntax.rs

```diff
@@ -0,0 +1,105 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{
+    Arguments, Expr, ExprCall, Keyword, Stmt, StmtExpr, StringLiteral, StringLiteralValue,
+};
+use ruff_python_semantic::SemanticModel;
+use ruff_source_file::Locator;
+
+use super::helpers::should_be_fstring;
+
+/// # What it does
+/// Checks for strings that contain f-string syntax but are not f-strings.
+///
+/// ## Why is this bad?
+/// An f-string missing an `f` at the beginning won't format anything, and instead
+/// treat the interpolation syntax as literal.
+///
+/// ## Example
+///
+/// ```python
+/// name = "Sarah"
+/// dayofweek = "Tuesday"
+/// msg = "Hello {name}! It is {dayofweek} today!"
+/// ```
+///
+/// It should instead be:
+/// ```python
+/// name = "Sarah"
+/// dayofweek = "Tuesday"
+/// msg = f"Hello {name}! It is {dayofweek} today!"
+/// ```
```

Comment:
The examples could probably be expanded. @charliermarsh would a description of the heuristics be useful, or is that too much?

Reply:
I think it'd make sense to include the heuristics as a standalone paragraph in the "Why is this bad?" , that's the pattern we tend to follow for other rules (e.g., to document when it does or does not trigger).